### PR TITLE
fix(build): inherit PyTorch C++ standard for hybrid EP

### LINKED ISSUE #698: `hybrid_ep_cpp` extension hardcodes `-std=c++17`; should inherit torch's C++ standard like `deep_ep_cpp`
**Describe the bug**

`DeepEP`'s `hybrid-ep` branch hardcodes `-std=c++17` for the `hybrid_ep_cpp` CUDAExtension in `setup.py` (`get_extension_hybrid_ep_cpp()`). Recent PyTorch builds require C++20 — their headers emit `#error C++20 or later compatible compiler is required to use PyTorch` when included from a C++17 translation unit. As a result, building `hybrid_ep_cpp` against a current-PyTorch environment fails at compile time.

The sibling `deep_ep_cpp` extension (`get_extension_deep_ep_cpp()`) sets **no** explicit `-std` flag — its `nvcc_flags = ['-O3', '-Xcompiler', '-O3']` — and therefore inherits torch's default standard (now c++20). That is why `deep_ep_cpp` still builds while `hybrid_ep_cpp` does not.

**To Reproduce**

1. Install a recent PyTorch whose headers require C++20 (e.g. a current nightly).
2. Build DeepEP `hybrid-ep` with the hybrid-EP multinode extension enabled:
   ```
   HYBRID_EP_MULTINODE=1 CUDA_HOME=/usr/local/cuda pip install --no-build-isolation .
   ```
3. Compilation of `hybrid_ep_cpp` aborts with the C++20 `#error` shown below.

**Expected behavior**

`hybrid_ep_cpp` builds against current PyTorch, the same way the sibling `deep_ep_cpp` extension already does.

**Actual behavior**

Building the `hybrid_ep_cpp` extension aborts at compile time. `nvcc` invokes the host compiler with `-std=c++17` for the `hybrid_ep_cpp` sources, which include PyTorch headers (`torch/all.h:5`, `ATen/ATen.h:5`); those headers hard-`#error` under C++17:

```
error: #error C++20 or later compatible compiler is required to use PyTorch.
error: #error C++20 or later compatible compiler is required to use ATen.
...
ERROR: Failed building wheel for deep_ep
```

The build log shows a mix of `-std=c++17` invocations (for `hybrid_ep_cpp`) and `-std=c++20` invocations (for the sibling `deep_ep_cpp`, which inherits torch's standard); only the C++17 ones fail. `pip install .` then exits with code 1, so DeepEP is not installed and every downstream consumer is blocked.

**Proposed fix**

Remove the hardcoded `-std=c++17` entry from the `hybrid_ep_cpp` `nvcc` compile args in `setup.py`, so the extension inherits torch's C++ standard from `CUDAExtension`/`BuildExtension` — exactly like `deep_ep_cpp`. This is portable: it builds against both C++17 torch (older) and C++20 torch (current), rather than swapping one hardcoded standard for another (a hardcoded `-std=c++20` would instead break users still on a C++17 torch). The runtime JIT compiler (`csrc/hybrid_ep/jit/compiler.cu`) can keep `-std=c++17`, since its JIT-compiled kernels do not include torch headers and it relies on `std::any`.

Validated experimentally: with the hardcoded flag removed, the extension compiles cleanly and a DeepSeek-V3 MXFP8 expert-parallel (EP=8) training run completes end-to-end with healthy convergence (loss 12.02 decreasing to 6.60 across 50 steps) and no runtime regressions on GB200-class hardware (CUDA arch `10.0`/`9.0`, PyTorch built for C++20).

**Environment**

- DeepEP: `hybrid-ep` branch, commit `17cfb817bccec3a9c247013360cc550c2bac441e`
- PyTorch: current build requiring C++20 (arm64/aarch64)
- CUDA arch list: `10.0 9.0`
- OS/arch: Linux aarch64

---

_This issue was drafted with assistance from the `opus` AI model._

### PR DESCRIPTION
**Pull request by tiramitree**
## Summary

- remove the hardcoded `-std=c++17` flag from `hybrid_ep_cpp`
- let PyTorch's `CUDAExtension` / `BuildExtension` select the compatible C++ standard, matching `deep_ep_cpp`
- add a standard-library regression test that rejects any exact hardcoded `-std=c++NN` string anywhere in this extension builder, including future appended or `cxx` flags

Fixes #698.

## Why

Recent PyTorch headers require C++20, while older supported PyTorch releases may still select C++17. Inheriting PyTorch's selected standard avoids forcing either generation and keeps the two DeepEP extensions consistent.

The runtime JIT compiler's separate C++17 flag is intentionally unchanged because this patch only covers the PyTorch-header extension build.

## Validation

- The regression scanner finds `-std=c++17` on the exact base, `hybrid-ep@94a9f8f6b146c07d97ec58f67cd6d303296d6098`.
- After the source change, `python -m unittest tests/test_setup_compile_flags.py -v` passes.
- The exact test also passes under an actual official Python 3.8.10 Windows embeddable runtime, covering the repository's documented minimum; it also passes under Python 3.12.10.
- `python -m py_compile setup.py tests/test_setup_compile_flags.py`
- `git diff --check`

## Validation boundary

I did not run an NVCC/CUDA build, GPU workload, or training run in this environment. This PR's local evidence is limited to the red/green structural regression, Python 3.8/3.12 test execution, Python compilation, and diff validation.

The implementation, test, review-response update, and PR text were prepared with material AI assistance and checked against the current source and issue before submission.

### CODE REVIEW

File: tests/test_setup_compile_flags.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import ast
+import re
+import unittest
+from pathlib import Path
+
+
+SETUP_PATH = Path(__file__).resolve().parents[1] / "setup.py"
+
+
+def get_literal_nvcc_flags(function_name: str) -> list[str]:
```
[the remark this answers is not in the document]
Comment (tiramitree) [reply tests/test_setup_compile_flags.py]: Fixed in fc7dc20. The test no longer uses built-in generic annotations or zip(strict=True), and the exact test now passes under the official Python 3.8.10 Windows embeddable runtime as well as Python 3.12.10.

File: tests/test_setup_compile_flags.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            if not isinstance(flags, list) or not all(isinstance(flag, str) for flag in flags):
+                raise AssertionError("compile_args['nvcc'] must be a literal list of strings")
+            return flags
+
+    raise AssertionError("compile_args does not define nvcc flags")
+
+
+class TestSetupCompileFlags(unittest.TestCase):
+    def test_hybrid_ep_inherits_torch_cxx_standard(self):
+        nvcc_flags = get_literal_nvcc_flags("get_extension_hybrid_ep_cpp")
```
[the remark this answers is not in the document]
Comment (tiramitree) [reply tests/test_setup_compile_flags.py]: Addressed in fc7dc20. The guard now walks every string constant inside get_extension_hybrid_ep_cpp and rejects any exact -std=c++NN flag, so it also catches future cxx entries and append/extend paths rather than only the initial nvcc list literal.

File: tests/test_setup_compile_flags.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        for node in ast.walk(function)
+        if isinstance(node, ast.Assign)
+        and any(
+            isinstance(target, ast.Name) and target.id == "compile_args"
+            for target in node.targets
+        )
+    )
+    if not isinstance(compile_args, ast.Dict):
+        raise AssertionError("compile_args must be a literal dictionary")
+
```
[the remark this answers is not in the document]
Comment (tiramitree) [reply tests/test_setup_compile_flags.py]: Confirmed and fixed in fc7dc20. The revised test uses Python 3.8-compatible syntax and passed under an actual Python 3.8.10 runtime.

**Review by thepointer1982 — APPROVED** [review]
